api_app/api/routes/workspaces.py

```diff
@@ -73,14 +73,19 @@ def _safe_get_workspace_role(user, workspace, user_role_assignments):
 
 
 @workspaces_core_router.get("/workspaces/{workspace_id}", response_model=WorkspaceInResponse, name=strings.API_GET_WORKSPACE_BY_ID)
-async def retrieve_workspace_by_workspace_id(user=Depends(get_current_tre_user_or_tre_admin), workspace=Depends(get_workspace_by_id_from_path)) -> WorkspaceInResponse:
-    access_service = get_access_service()
-    user_role_assignments = get_identity_role_assignments(user)
-    if access_service.get_workspace_role(user, workspace, user_role_assignments) != WorkspaceRole.NoRole:
+async def retrieve_workspace_by_workspace_id(request: Request, user=Depends(get_current_tre_user_or_tre_admin), workspace=Depends(get_workspace_by_id_from_path)) -> WorkspaceInResponse:
+    # check if the user is a TRE admin - if they are, return the workspace. If not, catch it and check their workspace role membership.
+    try:
+        user = await get_current_admin_user(request)
         return WorkspaceInResponse(workspace=workspace)
-    else:
-        logging.debug("User doesn't have roles in workspace.")
-        raise HTTPException(status_code=status.HTTP_403_FORBIDDEN, detail=strings.ACCESS_USER_IS_NOT_OWNER_OR_RESEARCHER)
+    except Exception:
+        access_service = get_access_service()
```

Comment:
Can this logic not be done using one of the other methods in `authentication.py` similar to `get_current_admin_user`? Say `get_current_workspace_owner_or_researcher_user_or_airlock_manager` ?

Reply:
yes, but - we need to allow any person to call the endpoint, so can't use a `depends` just to get the tre admin user?

Reply:
I meant instead of checking the user roles below, a second call could be used to one of the methods in `authentication.py` rather than a depends.

Or we could create a new entry in `authentication.py` that covers every user to use in a depends? Would make it clearer to read.

Also at the top of this file is already a dependency:

```
workspaces_core_router = APIRouter(dependencies=[Depends(get_current_tre_user_or_tre_admin)])
```
Will this cause an issue?

Reply:
The `get_current_xxx` methods check the roles by decoding the token passed to them from the call, so they'd need to be under the workspace router. In this instance we need to be under the main router to get the workspace details in the first place, so the method to get the role membership is by reaching out to AAD.